### ISSUE TITLE
Enrich Mediator - Error when source property's value contains colon

### ISSUE THREAD

**Issue by bsenduran** [wso2/product-ei#5353]
**Description:**
When the enrich mediator's source is from a property and its value contains a colon for example a time stamp. enrich mediator fails with error 
```
com.google.gson.JsonSyntaxException: com.google.gson.stream.MalformedJsonException: Use JsonReader.setLenient(true) to accept malformed JSON at line 1 column 4 path $
	at com.google.gson.JsonParser.parse(JsonParser.java:65)
	at com.google.gson.JsonParser.parse(JsonParser.java:45)
	at org.apache.synapse.mediators.elementary.EnrichMediator.mediate(EnrichMediator.java:132)
```

**Suggested Labels:**
Synapse, enrich mediator

**Affected Product Version:**
EI 6.5.0

**Steps to reproduce:**
Following is a sample API to reproduce this issue
```xml
<api xmlns="http://ws.apache.org/ns/synapse" name="enrichtest" context="/enrichtest">
   <resource methods="GET">
      <inSequence>
         <payloadFactory media-type="xml">
            <format>
               <time xmlns="">null</time>
            </format>
            <args/>
         </payloadFactory>
         <property name="newtime" value="10:10:00" scope="default" type="STRING"/>
         <enrich>
            <source type="property" clone="true" property="newtime"/>
            <target xpath="//time"/>
         </enrich>
         <respond/>
      </inSequence>
   </resource>
</api>
```
if the newtime property's value is changed to a string without colon this API works without any issue

**Related Issues:**
Following PR has a logic that if the source is not an OM element that is is treated as JSON string since the time value has colon the gson library thinks as  a malformed JSON
https://github.com/wso2/wso2-synapse/pull/1324

**Comment by arunans23** [comment]
This has been fixed in https://github.com/wso2/wso2-synapse/pull/1407 and the fix is available in EI 6.6.0.

**Comment by bsenduran** [comment]
Tested with EI 6.6.0 and the issue is fixed. Hence closing
